### DISCUSSION TITLE
Mounting subdirectory into container as user's files

### DISCUSSION THREAD

**Discussion by paul** [nextcloud/all-in-one#4909] [Questions] [answered]
I've started using this AIO setup, because it seems to be better supported and maintained than other nextcloud docker solution, so thanks for that. However, there's a use-case that I used to do, that I'm not sure how to accomplish in this AIO setup.

I have my personal "documents" folder, that's been around longer then nextcloud, that I've been sharing with my computers via smb, nfs, or whatever else. Its currently hosted on my fileserver as a zfs subvolume. When I started using nextcloud I symlinked the folder to `/var/www/html/data/{myself}/files`, and it worked. Later, when using the non-aio containers provided by truenas or whatever else, I was still able to mount that folder into the right place into the container. 

I'm not sure how to accomplish the same with the AIO container. I'm able to mount the data volume, which I've done, but I don't know how to mount my documents into that at the right place (`./data/{myself}/files`), with the right permissions. 

I've tried a symlink, and a bind mount on the host machine, but the nextcloud still shows the original files, not mine. I suspect it might be something with permissions, or maybe the docker volume isn't following the bind mount. 

In any case, is there any way to accomplish what I'm after? I'm hoping to avoid restructuring my subvolumes, or having a subdir within nextcloud be the "source of truth" for my documents folder.

**Comment by szaimen** [comment, marked as answer]
Hi, you should be able to achieve something comparable by setting [NEXTCLOUD_DATADIR to a path on your host](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-default-location-of-nextclouds-datadir) if not already done and then using a bind mount via fstab on your host in order to mount `./data/{myself}/files` into the coorect location inside this specified datadir.

**Reply by paul** [reply]
This seemed to have worked, kinda by coincidence. I had to `chown -R {myself}:www-data; chmod g+w`, and luckily the gid for www-data is the same on the host, and within the nextcloud container, so now it can read those files.
